Framework/Crystal/src/CalibrationHelpers.cpp

```diff
@@ -0,0 +1,165 @@
+    double tx = atan2(-Z[1], Z[2]);
+    double tz = atan2(-Y[0], X[0]);
+    double cosy = Z[2] / cos(tx);
+    double ty = atan2(Z[0], cosy);
+    Rotx = (tx * RAD_TO_DEG);
+    Roty = (ty * RAD_TO_DEG);
+    Rotz = (tz * RAD_TO_DEG);
+  } else // roty = 90 0r 270 def
+  {
+    double k = 1;
+    if (Z[0] < 0)
+      k = -1;
+    double roty = k * 90;
+    double rotx = 0;
+    double rotz = atan2(X[2], Y[2]);
+
+    Rotx = (rotx * RAD_TO_DEG);
+    Roty = (roty * RAD_TO_DEG);
+    Rotz = (rotz * RAD_TO_DEG);
+  }
+}
+
+/**
+ * Updates the ParameterMap for NewInstrument to reflect the position of the
+ *source.
+ *
+ * @param newInstrument The instrument whose parameter map will be changed to
+ *reflect the new source position
+ * @param L0 The distance from source to sample (should be positive)
+ * @param newSampPos The relative shift for the new sample position
+ * @param detectorInfo DetectorInfo for the workspace being updated
+ */
+void CalibrationHelpers::fixUpSampleAndSourcePositions(
+    boost::shared_ptr<const Instrument> newInstrument, double const L0,
+    const V3D newSampPos, DetectorInfo &detectorInfo) {
+  boost::shared_ptr<ParameterMap> pmap = newInstrument->getParameterMap();
+
+  IComponent_const_sptr source = newInstrument->getSource();
+  IComponent_const_sptr sample = newInstrument->getSample();
+
+  V3D samplePos = detectorInfo.samplePosition();
+  if (samplePos != newSampPos) {
+    detectorInfo.setPosition(*sample, newSampPos);
+  }
+  V3D sourceRelPos = source->getRelativePos();
+  V3D sourcePos = detectorInfo.sourcePosition();
+  V3D parentSourcePos = sourcePos - sourceRelPos;
+  V3D source2sampleDir = samplePos - source->getPos();
+
+  double scalee = L0 / source2sampleDir.norm();
+  V3D newsourcePos = newSampPos - source2sampleDir * scalee;
+  V3D newsourceRelPos = newsourcePos - parentSourcePos;
+
+  detectorInfo.setPosition(*source, newsourceRelPos);
+}
+
+/**
+ * Updates the ParameterMap for newInstrument to reflect the changes in the
+ *associated panel information
+ *
+ * @param bankNames The names of the banks (panels) that will be updated
+ * @param newInstrument The instrument whose parameter map will be changed to
+ *reflect the new values below
+ * @param pos The quantity to be added to the current relative position, from
+ *old NewInstrument, of the banks in bankNames.
+ * @param rot The quantity to be added to the current relative rotations, from
+ *old NewInstrument, of the banks in bankNames.
+ * @param detWScale The factor to multiply the current detector width, from old
+ *NewInstrument, by to get the new detector width for the banks in bankNames.
+ * @param detHtScale The factor to multiply the current detector height, from
+ *old NewInstrument, by to get the new detector height for the banks in
+ *bankNames.
+ * @param rotCenters Rotate the centers of the panels(the same amount) with the
+ *rotation of panels around their center
+ * @param detectorInfo DetectorInfo object for the
+ */
```

Comment:
I having trouble understanding what these `ComponentHelpers::fixUp...` methods do without having to read the implementation code, though, I understand that this was the naming used in the original code. I think we should consider renaming some of these.